injector/src/main/java/edu/ucr/cs/riple/injector/changes/RemoveAnnotation.java

```diff
@@ -32,11 +35,17 @@ public RemoveAnnotation(Location location, String annotation) {
   }
 
   @Override
-  public boolean apply(CompilationUnit tree) {
-    return this.location.apply(tree, annotation, false);
+  public void visit(NodeWithAnnotations<?> node) {
+    // We only insert annotations with their simple name, therefore, we should only remove
+    // the annotation if it matches with the simple name (otherwise, the annotation was not injected
+    // by the core module request and should not be touched).
+    AnnotationExpr annotationExpr = new MarkerAnnotationExpr(annotationSimpleName);
+    NodeList<AnnotationExpr> annotations = node.getAnnotations();
+    annotations.removeIf(annotationExpr::equals);
   }
 
   @Override
+  @SuppressWarnings("unchecked")
```

Comment:
This one should still be removed.

Reply:
dc472d03eebba3dd33485b776a0b98e35f155266